picasso/src/test/java/com/squareup/picasso/RequestCreatorTest.java

```diff
@@ -790,6 +797,28 @@ public void nullKeyTransformationInvalid() {
     });
   }
 
+  @Test(expected = IllegalArgumentException.class)
+  public void nullKeyInTransformationListInvalid() {
+    List<Transformation> transformations = new ArrayList<Transformation>();
+    transformations.add(new Transformation() {
+      @Override public Bitmap transform(Bitmap source) {
+        return source;
+      }
+
+      @Override public String key() {
+        return null;
+      }
+    });
+    new RequestCreator().transform(transformations);
+  }
+
+  @Test
+  public void transformationListImplementationValid() {
+    List<TestTransformation> transformations = new ArrayList<TestTransformation>();
+    transformations.add(new TestTransformation("test"));
+    new RequestCreator().transform(transformations);
```

Comment:
Ironically this test actually caught a bug when I wrote it originally.  I had forgot the `this.` in `this.transformations` in some code that's no longer part of the PR.  

I put a bit of thought into the quality of the tests.  There are no existing tests that checked if `.transform()` actually added the transformation.  `Request.Builder data` is a private field in `RequestCreator`, and `List<Transformation> transformations` in `Request.Builder` is as well,  and I assumed that you guys would be reluctant to increase visibility.  

But I'd definitely feel better about increasing visibility and writing proper tests.